### PR TITLE
feat:filter

### PR DESCRIPTION
**Pull request by tewarimohit**
Added Product Page:-
- Added Category Filter
- Added Range Filter
- Added Stars Filter.

https://user-images.githubusercontent.com/89827782/159937044-40314fff-3b24-4a19-9187-512bbbd66434.mp4

### CODE REVIEW

File: src/Frontend/Context/FilterContext.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	}, []);
+
+	const filteredProducts = Compose(
+		state,
+		getSortBy,
+		getRange,
+		filterCategory,
+		getRating
+	)([...products]);
+	// console.log("Filtered", filteredProducts);
```
Comment (SukanyaSen530) [inline src/Frontend/Context/FilterContext.js]: Remove it if not required .

**Review by SukanyaSen530 — APPROVED** [review]

File: src/Frontend/Components/Products/ProductComponents/FilterCompose.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+};
+const getRomanceBooks = (state, data) => {
+	return state.isRomance
+		? data.filter(item => item.categoryName === "romance")
+		: data;
+};
+const getShortStoryBooks = (state, data) => {
+	return state.isShortStory
+		? data.filter(item => item.categoryName === "short-story")
+		: data;
```
Comment (prathmeshjagtap) [inline src/Frontend/Components/Products/ProductComponents/FilterCompose.js]: you can remove this if you are not using

File: src/Frontend/Components/Products/ProductComponents/GetFilters.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+				isClassic: false,
+				isFiction: false,
+				isRomance: false,
+				isShortStory: false,
+				priceRange: 5000,
+				rateBy: "",
+			};
+
+		default:
+			console.log("Products not found.");
```
Comment (prathmeshjagtap) [inline src/Frontend/Components/Products/ProductComponents/GetFilters.js]: remove console.log

File: src/Frontend/Pages/Products.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import "../Styles/Home.css";
+import Header from "../Components/HeaderComponents/Header";
+import FilterArea from "../Components/Products/ProductComponents/FilterArea";
+import ProductsInfo from "../Components/Products/ProductComponents/ProductsInfo";
+import ProductCard from "../Components/Products/ProductComponents/ProductCard";
+import { useFilter } from "../Context/FilterContext.js";
+const Products = () => {
+	const { filteredProducts } = useFilter();
+
+	console.log("products", filteredProducts);
```
Comment (prathmeshjagtap) [inline src/Frontend/Pages/Products.js]: remove console.log

File: src/Frontend/Styles/Cart.css [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+	color: var(--ecom-login-bdr);
+	grid-template-columns: repeat(4, 1fr);
+	grid-gap: 10px;
+	grid-template-areas:
+		"head"
+		"side"
+		"info"
+		"content"
+		"footer";
+
```
Comment (prathmeshjagtap) [inline src/Frontend/Styles/Cart.css]: remove this

File: src/Frontend/Styles/Wishlist.css [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+	display: grid;
+	color: var(--ecom-login-bdr);
+	grid-template-columns: repeat(4, 1fr);
+	grid-gap: 10px;
+	grid-template-areas:
+		"head"
+		"info"
+		"content"
+		"footer";
+
```
Comment (prathmeshjagtap) [inline src/Frontend/Styles/Wishlist.css]: you can remove this code

**Review by prathmeshjagtap — CHANGES_REQUESTED** [review]

**Review by prathmeshjagtap — APPROVED** [review]

**Pull request merged**